src/script/microbit-interfacing/MicrobitSerial.ts

```diff
@@ -94,33 +76,68 @@ class MicrobitSerial implements MicrobitConnection {
         }
       });
     };
-    await this.usb.startSerial(processMessage);
     try {
+      await this.usb.startSerial(processMessage, handleError);
       await this.handshake();
+      stateOnConnected(DeviceRequestStates.INPUT);
 
       // Set the radio frequency to a value unique to this session
       const radioFreqCommand = protocol.generateCmdRadioFrequency(
-        MicrobitSerial.sessionRadioFrequency,
+        this.sessionRadioFrequency,
       );
       const radioFreqResponse = await this.sendCmdWaitResponse(radioFreqCommand);
-      if (radioFreqResponse.value !== MicrobitSerial.sessionRadioFrequency) {
+      if (radioFreqResponse.value !== this.sessionRadioFrequency) {
         throw new Error(
-          `Failed to set radio frequency. Expected ${MicrobitSerial.sessionRadioFrequency}, got ${radioFreqResponse.value}`,
+          `Failed to set radio frequency. Expected ${this.sessionRadioFrequency}, got ${radioFreqResponse.value}`,
         );
       }
 
-      // Request the micro:bit to start sending the periodic messages
-      const startCmd = protocol.generateCmdStart({
-        accelerometer: true,
-        buttons: true,
-      });
-      await this.usb.serialWrite(startCmd.message);
+      // For now we only support input state.
+      if (states.includes(DeviceRequestStates.INPUT)) {
+        // Request the micro:bit to start sending the periodic messages
+        const startCmd = protocol.generateCmdStart({
+          accelerometer: true,
+          buttons: true,
+        });
+        await this.usb.serialWrite(startCmd.message);
+        await this.sendCmdWaitResponse(startCmd);
+      }
+
+      stateOnAssigned(DeviceRequestStates.INPUT, this.usb.getModelNumber());
+      stateOnReady(DeviceRequestStates.INPUT);
     } catch (e) {
-      this.usb.stopSerial();
+      stateOnFailedToConnect(DeviceRequestStates.INPUT);
+      await this.usb.stopSerial();
       throw e;
     }
   }
 
+  async disconnect(userDisconnect: boolean): Promise<void> {
+    // We might want to send command to stop streaming here?
+    this.responseMap.clear();
+    await this.usb.stopSerial();
+    stateOnDisconnected(DeviceRequestStates.INPUT, userDisconnect);
+  }
+
+  async reconnect(userTriggered: boolean): Promise<void> {
+    await this.disconnect(userTriggered);
```

Comment:
We removed the disconnect call from the bluetooth implementation. We only call reconnect when we know we're disconnected.